sycl/plugins/hip/pi_hip.cpp

```diff
@@ -1637,8 +1637,15 @@ pi_result hip_piDeviceGetInfo(pi_device device, pi_device_info param_name,
     return getInfo(param_value_size, param_value, param_value_size_ret, value);
   }
 
-  // TODO: Implement.
-  case PI_DEVICE_INFO_ATOMIC_64:
+  case PI_DEVICE_INFO_ATOMIC_64: {
+    hipDeviceProp_t props;
+    cl::sycl::detail::pi::assertion(
+          hipGetDeviceProperties(&props, device->get()) == hipSuccess);
+    bool atomic64 = (props.arch.hasFloatAtomicAdd == 1) ? true : false;
```

Comment:
According to the [ROCm documentation](https://rocm-developer-tools.github.io/HIP/structhipDeviceArch__t.html):

<html>
<body>
<!--StartFragment-->

unsigned | hasFloatAtomicAdd: 1
-- | --
  | 32-bit float atomic add in global and shared memory.

<!--EndFragment-->
</body>
</html>

As such, this does not seem like the device property to check. Maybe `hasGlobalInt64Atomics` and maybe `hasSharedInt64Atomics` would be more appropriate?

Ping @AidanBeltonS & @AerialMantis comment.

Reply:
If the developers of the HIP plugin have the right property, please implement them now.  Thanks.